app/src/main/java/com/kickstarter/libs/utils/RewardViewUtils.kt

```diff
@@ -22,9 +22,15 @@ object RewardViewUtils {
      */
     @StringRes
     fun pledgeButtonText(project: Project, reward: Reward): Int {
+        val hasAddOnsSelected = project.backing()?.addOns()?.isNotEmpty() ?: false
+        val hasUnavailableAddOn:Boolean = project.backing()?.addOns()?.firstOrNull {
+            !RewardUtils.isAvailable(project, it)
+        }?.let { true } ?: false
```

Comment:
Would love to go over this with you to understand what exactly is happening here.

Reply:
In there I'm checking is any of the addOns is unavailable.
- backing.addOns is an optional list of rewards
- I use firstOrNull so it wont iterate over the entire list in case some addOn is unavailable, if there is some unavailable addOn on the let block return true. Null in case all the addOns are available so with the elvis operator `?:` sets the `hasUnavailableAddOn` to false